src/parser.rs

```diff
@@ -98,7 +105,7 @@ pub fn check_chunk_range(chunks: &Vec<EncodedChunk>) -> Result<(), RestoreError>
             .cloned()
             .collect::<Vec<u16>>();
         return Err(RestoreError::MissingChunk {
-            expected_total: expected_total,
+            expected_total,
```

Comment:
Is this working because of positional argument  ("expected total is first arg in struct and first arg here") or because the words match "expected_total is the same word for the struct field so it matches")

Reply:
https://doc.rust-lang.org/edition-guide/rust-2018/data-types/field-init-shorthand.html